apps/deploy-web/src/hoc/guard/guard.hoc.tsx

```diff
@@ -1,8 +1,18 @@
+import { Loading } from "@src/components/layout/Layout";
 import FourOhFour from "@src/pages/404";
 
-export const Guard = <P extends object>(Component: React.ComponentType<P>, useCheck: () => boolean, FallbackComponent = FourOhFour) => {
+type UseCheck = () => {
+  canVisit: boolean;
+  isLoading: boolean;
+};
+
+export const Guard = <P extends object>(Component: React.ComponentType<P>, useCheck: UseCheck, FallbackComponent = FourOhFour) => {
   const WithGuard = (props: P) => {
-    const canVisit = useCheck();
+    const { canVisit, isLoading } = useCheck();
+
+    if (isLoading) {
+      return <Loading text="Please wait..." />;
```

Comment:
question(non-blocking): Is there a way we can customize the text in different pages?